src/webgpu/util/device_pool.ts

```diff
@@ -69,7 +70,20 @@ export class DevicePool {
           holder.device.destroy();
         }
       }
-      throw ex;
+      // In the try block, we may throw an error if the device is lost, however, we want to suppress
+      // the error if the device lost was expected. The device lost is expected when
+      // `holder.expectedLostReason` is set to null or equal to `holder.lostInfo.reason`. The
+      // condition below is just the negation to avoid extra branching.
+      if (
+        !(
+          holder.lostInfo !== undefined &&
+          holder.expectedLostReason !== undefined &&
+          (holder.expectedLostReason === null ||
+            holder.expectedLostReason === holder.lostInfo.reason)
+        )
+      ) {
```

Comment:
Applied readability change. Note that the reason I am doing the assert is because the catch block marks the device for reinitialization for future tests. Otherwise, I would need to duplicate the device marking code. Since a destroyed device always needs to be reinitialized, I handle the special cases for suppressing the error in the catch block.